src/main/java/appeng/fluids/storage/FluidCellInventory.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2018, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.fluids.storage;
+
+
+import java.util.HashSet;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.fluids.FluidStack;
+
+import appeng.api.AEApi;
+import appeng.api.config.Actionable;
+import appeng.api.exceptions.AppEngException;
+import appeng.api.implementations.items.IStorageCell;
+import appeng.api.networking.security.IActionSource;
+import appeng.api.storage.IMEInventoryHandler;
+import appeng.api.storage.ISaveProvider;
+import appeng.api.storage.IStorageChannel;
+import appeng.api.storage.channels.IFluidStorageChannel;
+import appeng.api.storage.data.IAEFluidStack;
+import appeng.core.AEConfig;
+import appeng.core.AELog;
+import appeng.me.storage.CellInventoryBase;
+import appeng.me.storage.CellInventoryHandler;
+import appeng.util.item.AEFluidStack;
+
+
+/**
+ * @author DrummerMC
+ * @version rv6 - 2018-01-16
+ * @since rv6 2018-01-16
+ */
+public class FluidCellInventory extends CellInventoryBase<IAEFluidStack>
+{
+	private static final HashSet<String> BLACK_LIST = new HashSet<>();
```

Comment:
We should remove the blacklist completely. It is mostly obsolete since custom storage channels and it was mostly intended to allow addons to blacklist fake itemstacks for other types. Similar to AE1 and EC1.

Or alternatively it should be extended at some point, but then include API endpoints to allow blacklisting items like containers to avoid NBT/packet corruption due to size limits.